### PR TITLE
Add Fall 2019 Lab b1

### PR DESCRIPTION
**Pull request by BernardZhao**

### CODE REVIEW

File: labs/b1.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+### General Questions
+
+First we'll do some general questions to get you comfortable with working in the shell before we get to the real lab assignment. Feel free to use Google and work in a terminal (where applicable) to verify your conjectures.
+
+1. What differentiates Linux/OSX from operating systems like Windows?
+
+2. What are some differences between the command line and normal (graphical) usage of an OS?
+
+3. What is the path to root in linux?
```
Comment (mdcha) [inline labs/b1.md]: "What is the root directory in Linux"

Comment (abizer) [inline labs/b1.md]: this is going to lead to a lot of people typing `/root` instead of `/`

File: labs/b1.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+4. How can you use a command on a file not in a folder in your current directory?
```
Comment (abizer) [inline labs/b1.md]: these questions are pedagogically quite confusing. `How can you use a command on a file not in your current directory?`

File: labs/b1.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+10. What command modifies permissions? (multiple choice: chmod, ls, mkdir, nano)
+
+11. Which letter is short for the flag corresponding to the permission to write? (multiple choice: x, w, r)
+
+12. What would the contents of cat.txt be after running "echo 'hello'> cat.txt"
+
+13. What about now after I also run "echo 'world' >> cat.txt"
+
+14. What is the difference between '>' and '>>'?
```
Comment (mdcha) [inline labs/b1.md]: This belongs in the interactive portion

File: labs/b1.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+1. What differentiates Linux/OSX from operating systems like Windows?
+
+2. What are some differences between the command line and normal (graphical) usage of an OS?
+
+3. What is the path to root in linux?
+
+4. How can you use a command on a file not in a folder in your current directory?
+
+5. Give a command that can edit a file in root from anywhere in your file system.
```
Comment (mdcha) [inline labs/b1.md]: These should be multiple choice in the form of "You're currently in directory X. Which of these commands do Y to file Z?"

Comment (abizer) [inline labs/b1.md]: Even I would have a hard time interpreting what this question is asking, and what the correct response would be. Do you want them to demonstrate understanding of absolute paths? Roll this into the previous question and clarify it as
`Give an example of a command that runs on an arbitrary file located anywhere on the filesystem, regardless of your current working directory.` or something like that.

File: labs/b1.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+13. What about now after I also run "echo 'world' >> cat.txt"
+
+14. What is the difference between '>' and '>>'?
+
+### Culture Questions
+
+Next are some cultural questions about Linux and Open Source Software.
+
+1. What is the best text editor?(multiple choice: nano, vim, emacs, Notepad)
```
Comment (mdcha) [inline labs/b1.md]: Meme question

File: labs/b1.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+2. What are some differences between the command line and normal (graphical) usage of an OS?
+
+3. What is the path to root in linux?
+
+4. How can you use a command on a file not in a folder in your current directory?
+
+5. Give a command that can edit a file in root from anywhere in your file system.
+
+6. What are 1 or 2 options/flags and what do they do?
```
Comment (abizer) [inline labs/b1.md]: options/flags to what?

File: labs/b1.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+2. What's the difference between permissive and copyleft licenses?
+
+3. Give an example of a permissive license.
+
+4. Give an example of (1) open-source software and (2) free, but closed-source software.
+
+5. Name three different Linux distros
+
+6. What does FOSS stand for?
```
Comment (mdcha) [inline labs/b1.md]: 6-8 aren't pedagogically useful

File: labs/b1.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+4. How can you use a command on a file not in a folder in your current directory?
+
+5. Give a command that can edit a file in root from anywhere in your file system.
+
+6. What are 1 or 2 options/flags and what do they do?
+
+7. How would you create a copy of a file (multiple possible answers)?
+
+8. Which command deletes?
```
Comment (abizer) [inline labs/b1.md]: complete sentences? `What is the common command used to delete files`

File: labs/b1.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+9. How do you make it delete folders?
+
+10. What command modifies permissions? (multiple choice: chmod, ls, mkdir, nano)
```
Comment (abizer) [inline labs/b1.md]: are file permissions explained such that this question has context?

File: labs/b1.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+
+11. Which letter is short for the flag corresponding to the permission to write? (multiple choice: x, w, r)
```
Comment (abizer) [inline labs/b1.md]: delete this question

File: labs/b1.md (hunk 3 above)
Comment (abizer) [inline labs/b1.md]: There's not enough context in the previous two questions to adequately intuit that `>` (over)writes and `>>` appends, without having another question that has them `echo yes > cat.txt` afterwards.

File: labs/b1.md (hunk 5 above)
Comment (abizer) [inline labs/b1.md]: space after the question mark?

File: labs/b1.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+
+4. Give an example of (1) open-source software and (2) free, but closed-source software.
+
+5. Name three different Linux distros
+
+6. What does FOSS stand for?
+
+7. What was Linus's ethical concerns with some software?
+
+8. Who was Linus Torvalds?
```
Comment (abizer) [inline labs/b1.md]: delete these last two questions, they're irrelevant

File: labs/b1.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+   to find in there! Go find it without actually opening up the file itself! 
+   (Hint: It's a url)
+
+   Two lines above the URL is a secret solution. What is that solution?
+   Hint for this one: "Context Line Control"
+
+7. Finally, there's an empty file called `hello_world` in the directory. Write
+   your name in it!
+
+8. Finally, **remember to submit your answers to gradescope**. There's some other questions
```
Comment (abizer) [inline labs/b1.md]: Gradescope and link to the class

**Pull request merged**